vendor/github.com/labstack/echo/middleware/auth.go

```diff
@@ -0,0 +1,49 @@
+package middleware
+
+import (
+	"encoding/base64"
+	"net/http"
+
+	"github.com/labstack/echo"
+)
+
+type (
+	BasicValidateFunc func(string, string) bool
+)
+
+const (
+	Basic = "Basic"
+)
+
+// BasicAuth returns an HTTP basic authentication middleware.
+//
+// For valid credentials it calls the next handler.
+// For invalid credentials, it sends "401 - Unauthorized" response.
+func BasicAuth(fn BasicValidateFunc) echo.HandlerFunc {
+	return func(c *echo.Context) error {
+		// Skip WebSocket
+		if (c.Request().Header.Get(echo.Upgrade)) == echo.WebSocket {
+			return nil
+		}
+
+		auth := c.Request().Header.Get(echo.Authorization)
+		l := len(Basic)
+
+		if len(auth) > l+1 && auth[:l] == Basic {
+			b, err := base64.StdEncoding.DecodeString(auth[l+1:])
+			if err == nil {
+				cred := string(b)
+				for i := 0; i < len(cred); i++ {
+					if cred[i] == ':' {
+						// Verify credentials
+						if fn(cred[:i], cred[i+1:]) {
+							return nil
+						}
+					}
+				}
```

Comment:
BTW-- why did we get just these four echo middlewares?  Seems odd -- especially since I do not see any change to GoDep in this PR.  I'd expect to see changes in vendor triggered by changes in the GoDep configuration.